benchmark/components/perf_fec_base.h

```diff
@@ -0,0 +1,552 @@
+    void create_files(
+        std::vector<std::string>& files,
+        size_t bytes,
+        bool rand_fill = true)
+    {
+        std::vector<uint8_t> buf(bytes);
+        const size_t files_nb = files.size();
+
+        if (rand_fill) {
+            for (size_t i = 0; i < files_nb; ++i) {
+                this->randomize_buffer(buf.data(), bytes);
+
+                const std::string file_name = files[i];
+                std::ofstream outfile(file_name, std::ofstream::binary);
+                outfile.write(reinterpret_cast<char*>(buf.data()), bytes);
+                outfile.close();
+            }
+        } else {
+            for (size_t i = 0; i < files_nb; ++i) {
+                const std::string file_name = files[i];
+                std::ofstream outfile(file_name, std::ofstream::binary);
+                outfile.seekp(bytes);
+                outfile << 'a';
+                outfile.close();
+            }
+        }
+    }
+
+    void create_files_name(
+        const std::string& prefix,
+        const std::string& suffix,
+        size_t files_nb,
+        std::vector<std::string>& files)
+    {
+        files.reserve(files_nb);
+        for (size_t i = 0; i < files_nb; ++i) {
+            std::string file_name =
+                prefix + std::string("_") + std::to_string(i) + suffix;
+            files.push_back(file_name);
+        }
+    }
+
+    void prepare_codec_streams()
+    {
+        const size_t coded_files_nb = MAX_CODE_LEN;
+        const size_t data_files_nb =
+            coded_files_nb * (MAX_INV_RATE - 1) / MAX_INV_RATE;
+        const size_t bytes = fragment_size;
+
+        create_files_name("data", ".dat", data_files_nb, d_files);
+        create_files_name("coded", ".dat", coded_files_nb, c_files);
+        create_files_name("recovered", ".dat", data_files_nb, r_files);
+
+        if (do_create_files) {
+            create_files(d_files, bytes);
+            create_files(c_files, bytes, false /* not fill random value */);
+            create_files(r_files, bytes, false /* not fill random value */);
+            do_create_files = false;
+        }
+    }
+
+    void open_streams(
+        const std::vector<std::string>& files,
+        std::vector<std::istream*>& streams,
+        bool in_and_out_flag = false)
+    {
+        std::ios_base::openmode mode = std::ios_base::in;
+        if (in_and_out_flag) {
+            mode |= std::ios_base::out;
+        }
+        for (size_t i = 0; i < streams.size(); ++i) {
+            streams[i] = new std::ifstream(files[i], mode);
+        }
+    }
+
+    void open_streams(
+        const std::vector<std::string>& files,
+        std::vector<std::ostream*>& streams,
+        bool in_and_out_flag = false)
+    {
+        std::ios_base::openmode mode = std::ios_base::out;
+        if (in_and_out_flag) {
+            mode |= std::ios_base::in;
+        }
+        for (size_t i = 0; i < streams.size(); ++i) {
+            streams[i] = new std::ofstream(files[i], mode);
```

Comment:
can you use unique_ptr here?

Reply:
yes, we can :)

Reply:
unfortunately, in [our API](https://github.com/scality/quadiron/blob/master/src/fec_base.h#L185) we use `std::vector<std::istream*>`. I can use `unique_ptr` here but it adds complexities, either create a new vector of pointers for API, or modify the API to use `unique_ptr` that will do big changes.
So I prefer to keep it for now.

Reply:
Ok, fair enough